test/puppetlabs/jdbc_util/pool_test.clj

```diff
@@ -140,8 +141,29 @@
   (let [test-pool (-> core-test/test-db
                       pool/spec->hikari-options
                       pool/options->hikari-config
-                      (pool/connection-pool-with-delayed-init identity 5000))]
+                      (pool/connection-pool-with-delayed-init {:migration-db core-test/test-db} identity 5000))]
     (.getConnection test-pool)
     (is (= {:state :ready}
            (pool/status test-pool)))
     (.close test-pool)))
+
+(deftest migration-db-spec-test
```

Comment:
Ahh I get now that this is only testing that the migration user is being used but seeing the exception was a little alarming.